docs: react-tinacms-editor for developers (#1310)
tinacms/tinacms: packages/react-tinacms-editor/editor-docs/README.md

5. [History](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/plugins/History)
6. [Image](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/plugins/Image)
7. [Inline](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/plugins/Inline)
8. [Link](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/plugins/Link)
9. [List](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/plugins/List)
10. [Table](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/plugins/Table)

Typical structure of a plugin:

- Menu
  - MarkdownMenu
  - ProsemirrorMenu
- Popups
  - MyPopup,...
- nodeView
- plugin
- commands

`Menu` folder has 2 components `MarkdownMenu` and `ProsemirrorMenu`, if present these are added to menubar in Markdown mode and Prosemirror mode respectively.

`Popups` folder has various Popup components. For instance image and link have Popup components used for editing. These components should take care for show / hide themselves conditionally.

> `nodeView` is used to create customized vide for prosemirror node, this is used for code block and image nodes.

Reviewer: ```suggestion
`nodeView` is used to create a custom view for the prosemirror node. This is used for code block and image nodes.
```

> `plugin` is prosemirror [plugin](https://prosemirror.net/docs/ref/#state.Plugin_System) object. It helps maintain state for plugin for instance for links and images it helps to keep information when user is currently editing link or image. It also enables other functionality like attaching event listeners, creation editor decorations, etc.

Reviewer: ```suggestion
`plugin` is prosemirror [plugin](https://prosemirror.net/docs/ref/#state.Plugin_System) object. It helps maintain state for plugins, such as whether or not the user is currently editing a link or image. The `plugin` also enables other functionality like attaching event listeners, creation editor decorations, etc.
```

> `commands` keeping commands related to the plugin together.

Reviewer: ```suggestion
`commands` defines the [prosemirror commands](https://prosemirror.net/docs/ref/#commands) related to the plugin.
```

> Currently all the `keyboard shortcuts` and prosemirror `input rules` are combined in `common plugin` it is required to refactor these into their respective plugins.

Reviewer: ```suggestion
Currently all the [keyboard shortcuts](https://prosemirror.net/docs/ref/#keymap) and [input rules](https://prosemirror.net/docs/ref/#inputrules) are combined in the `common plugin`. Future work will move these into their respective plugins.
```

### Markdown Editor

> [This](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/components/MarkdownEditor) editor is simple textarea where editor content can be edited a plain text. Currently menubar options are not enabled in this editor.

Reviewer: ```suggestion
The [Markdown Editor](https://github.com/tinacms/tinacms/tree/master/packages/react-tinacms-editor/src/components/MarkdownEditor) is simple textarea where editor content can be edited a plain text. Currently menubar options are not enabled in this editor.
```
